diff --git a/grovel/grovel.lisp b/grovel/grovel.lisp
index 1a5852f..62dc73f 100644
--- a/grovel/grovel.lisp
+++ b/grovel/grovel.lisp
@@ -245,7 +245,8 @@ int main(int argc, char**argv) {
 
 (defparameter *cc*
   #+(or cygwin (not windows)) "cc"
-  #+(and windows (not cygwin)) "c:/msys/1.0/bin/gcc.exe")
+  #+(and windows (not ecl) (not cygwin)) "c:/msys/1.0/bin/gcc.exe"
+  #+(and windows ecl (not cygwin)) c:*cc*)
 
 (defparameter *cc-flags*
   (append
